### PR TITLE
Adds remaining OpInfos and removes redundant test generators

### PR DESCRIPTION
**Pull request by mruberry**
Per title.

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3239,6 +3290,32 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_out=False,
            sample_inputs_func=sample_inputs_max_min_reduction_no_dim,),
+    OpInfo('median',
```
Comment (imaginary-person) [inline torch/testing/_internal/common_methods_invocations.py:3293]: ROCm `test_unsupported_dtypes` might probably fail, as `dtypesIfROCm` would be `floating_types()`, by default.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: That's a good concern but we skip test_unsupported_dtypes on ROCm:

```
00:12:57   test_unsupported_dtypes_var_mean_cuda_bfloat16 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_bool (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_complex128 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_complex64 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_float16 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_int16 (__main__.TestOpInfoCUDA) ... skip (0.049s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_int32 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_int64 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_int8 (__main__.TestOpInfoCUDA) ... skip (0.050s)
00:12:57   test_unsupported_dtypes_var_mean_cuda_uint8 (__main__.TestOpInfoCUDA) ... skip (0.051s)
```

https://github.com/pytorch/pytorch/blob/ec38dda1cce88400c44793dade881bef86dc447a/test/test_ops.py#L33

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Thanks for the clarification, @mruberry! Could there be any other tests that are now being tested on fewer dtypes on ROCm?

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: ROCm inherits from dtypes:

https://github.com/pytorch/pytorch/blob/ec38dda1cce88400c44793dade881bef86dc447a/torch/testing/_internal/common_methods_invocations.py#L187

So unless someone explicitly sets dtypes it will just test float and double. We thought about switching it to inherit from dtypesIfCUDA, which is set more often and would expand its dtypes, but that would require someone take the time to hammer the ROCm CI a bit.

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Thanks for the info, @mruberry!
BTW, some tests use the `@dtypesIfCUDA` decorator. Are these tests also run on AMD GPUs, unless the @skipCUDAIfRocm decorator is used?

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: > but that would require someone take the time to hammer the ROCm CI a bit

I can try :)

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: I appreciate that, @imaginary-person, but I should have added that this test extension is not a clear win. While more test coverage is generally good, it would also increase the time the CI takes to run and may degrade CI signal quality. We should probably focus on other issues and not make such a significant change to the current CI surface.

(If you'd like an issue suggestion please let me know!)

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Thank you, @mruberry!
Please suggest an issue I can work on.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: One of the sub-issues from the bigger OpInfo porting issue is handling the binary comparison operators: https://github.com/pytorch/pytorch/issues/54300. I know you're already familiar with OpInfos, but since we're getting closer to finishing them porting the ops in this class would be very helpful. Porting them together should save time, too, since they should be able to share sample inputs and have similar properties.

If that doesn't sound interesting, or you're looking for another issue after #54300, then there are some operators with bugs or missing features, too. Let me know what you think.

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Hello @mruberry, thanks for your response! I'd submit a PR for #54300 by tomorrow afternoon, if that works.

Thanks!

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: No rush but that'd be great!

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Hello @mruberry, please let me know if some sample inputs of `eq`, `equal`, and `ne` be made equal using `random.seed()`, or if the elements of the tensors wouldn't matter for `OpInfo` tests. Thanks!

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Great question, @imaginary-person. I wouldn't set the seed but instead create interesting combinations directly, by hand, as fixtures.

Comment (imaginary-person) [reply torch/testing/_internal/common_methods_invocations.py:3293]: Thanks for the clarification, @mruberry! I'll add such sample inputs in #55709 in a while.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1053,6 +1053,57 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
                                           requires_grad=requires_grad),))
     return inputs
 
+# Generates input tensors for testing reduction ops
+def _generate_reduction_inputs(device, dtype, requires_grad):
+    yield make_tensor((), device, dtype, requires_grad=requires_grad)
+    yield make_tensor((2,), device, dtype, requires_grad=requires_grad)
+    yield make_tensor((2, 3), device, dtype, requires_grad=requires_grad, discontiguous=True)
```
Comment (heitorschueroff) [inline torch/testing/_internal/common_methods_invocations.py:1060]: discontiguous -> noncontiguous

**Review by heitorschueroff — APPROVED** [review]
Needs to rebase and fix conflicts. Also, update the make_tensor parameter to `noncontiguous`. Otherwise LGTM.
